pkg/slo-controller/config/colocation_cm_event_handler.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/koordinator-sh/koordinator/pkg/util"
+)
+
+var _ handler.EventHandler = &ColocationHandlerForConfigMapEvent{}
+
+type ColocationCfgCache interface {
+	GetCfgCopy() *ColocationCfg
+	IsAvailable() bool
+	IsErrorStatus() bool
+}
+
+type colocationCfgCache struct {
+	lock          sync.RWMutex
+	colocationCfg ColocationCfg
+	available     bool
+	errorStatus   bool
```

Comment:
how about store the error itself

Reply:
We don't need to save the error message because it's already logged